src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,507 @@ title: Builder.io & Astro
+- package.json
+</FileTree>
+
+
+## Making a blog with Astro and Builder
+
+With the set up above, you can now create a blog with Astro and Builder.
+
+### Creating a model for a blog post
+
+
+In the Builder app create the model that will represent a blog post: go to the **Models** tab and click the **Create Model** button to create model with the following fields and values:
+
+- **Type:** Section
+- **Name:** "blogpost"
+- **Description:** "This model is for a blog post"
+
+:::note[Why a section model?]
+In a section model the content can sit between other elements of a page - like a `<header>` and `<footer>`, for example.
+:::
+
+In your new model use the **New Custom Field** button to create 2 new fields:
+
+1. Text field
+    - **Name:** "title"
+    - **Required:** Yes
+    - **Default value:** "I forgot to give this a title"
+
+    (leave the other parameters as their defaults)
+
+2. Text field
+    - **Name:** "slug"
+    - **Required:** Yes
+    - **Default value:** "some-slugs-take-their-time"
+
+    (leave the other parameters as their defaults)
+
+Then click the **Save** button in the upper right. 
+
+And put the name of the model in the `.env` file for convenience:
+```ini title=".env" ins={2}
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+BUILDER_BLOGPOST_MODEL='blogpost'
+```
+
+:::caution
+There are some pitfalls with the `slug` field:
+
+* Make sure your slug is not just a number. This seems to break the fetch request to Builder's API. 
+* Make sure your slugs are unique, since your site's routing will depend on that.
+:::
+
+### Setting up the preview
+
+To use Builder's visual editor, create the page `src/pages/builder-preview.astro` that will render the special `<builder-component>`:
+
+<FileTree title="Project Structure">
+- src/
+  - pages/
+    - **builder-preview.astro**
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Then add the following content:
+
+```astro title="src/pages/builder-preview.astro"
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+---
+
+<html lang="en">
+  <head>
+    <title>Preview for builder.io</title>
+  </head>
+  <body>
+    <header>This is your header</header>
+
+    <builder-component model={builderModel} api-key={builderAPIpublicKey}
+    ></builder-component>
+    <script async src="https://cdn.builder.io/js/webcomponents"></script>
+
+    <footer>This is your footer</footer>
+  </body>
+</html>
+
+```
+
+`<builder-component>` tells Builder where to insert the content from its CMS - it represents the portion of the page that's editable in the Builder app. In this guide it stands for the content of a blog post. The whole `builder-preview.astro` page recreates how a blog post will be embedded into the site later on - sitting between a `<header>` and a `<footer>`.
+
+The `<script>` tag here won't impact your site's performance, since nobody will visit it, except for developers.
```

Comment:
```suggestion
In the above example, `<builder-component>` tells Builder where to insert the content from its CMS.
```
This felt a little repetative and over-explainy. I think this will do!

Reply:
I made this extra explainy, because I felt that the `<builder-component>` is so crucial and unusual. It was also important to me to ease the reader into the concept of the preview url, which also - took me some effort to wrap my mind around. For example, I think I would wonder about why this page contains a `<header>` and a `<footer>`, if it wasn't explained explicitly.

Reply:
This just feels like the Astro concept of a slot to me, where content is injected somewhere: in this case, the content coming from the Builder CMS is injected into the `<builder-component>`. 

I think it does a disservice to someone trying to follow instructions and complete the task to have to read:
- this component is where the Builder content is inserted
- it represents the content from Builder
- it represents your blog post content from builder
- this entire page is an Astro layout, that works exactly like an Astro layout does, like how your content is a different component from your header and footer.
... none of which is actually a step to follow or a thing to do.

Nothing about a recipe should be extra explainy! That's what reference docs are for. These are steps to follow, with the minimum commentary so they don't get tripped up.  :)

To me, the helpful thing to call out is that `<builder-component>` is your slot for content.

Reply:
I feel like I'm getting the vibe now. That actually helps a lot in understanding your concept of a guide. 

> Nothing about a recipe should be extra explainy! That's what reference docs are for. These are steps to follow, with the minimum commentary so they don't get tripped up. :)

I was definitely writing from a more explainy place than that :)

Marking as resolved, go ahead!